src/lib-components/SectionPagination.vue

```diff
@@ -1,116 +1,189 @@
-    default: '',
+    required: false,
   },
   previousTo: {
     type: String,
-    default: '',
+    required: false,
+  },
+  pages: {
+    type: Number,
+    required: false,
+  },
+  initialCurrentPage: {
+    type: Number,
+    required: false,
   },
 })
+const emit = defineEmits(['changePage']) // let parent component know when page changes
+const theme = useTheme()
+const maxPages = ref(10) // default # of buttons that will fit in container, gets recalculated onMount & resize
+const leftPages = ref([33]) // an array of numbers representing the page buttons that will appear ( we start with a single '33' so we can measure the width of a button to calc maxPages)
+const currPage = ref(1) // current page, defaults to 1
+const pageButtons: Ref<HTMLElement | null> = ref(null)
+
+// METHODS
+function handlePageChange(item: number) {
+  if (initialCurrentPage && pages) {
+    if (currPage.value !== item) {
+      currPage.value = item
+      generateLeftPages()
+      emit('changePage', item)
+    }
+  }
+}
+function generateLeftPages() {
+  if (pages && maxPages) {
+    let start = 1
+    // stop at either maxPages or total pages, whichever is lesser
+    let stop = Math.min(maxPages.value, pages)
+
+    // if current page is greater than than maxPages,
+    // put current page in middle of range of generated page number buttons
+    if (currPage.value > maxPages.value) {
+      let newMaxPages = maxPages.value - 4 // subtract 4 for '...' first/last number buttons
+      start = Math.max(1, currPage.value - Math.floor(newMaxPages / 2))
+      stop = start + newMaxPages
+
+      // if current page is very near the last page,
+      // we need to remove the truncation button near the end
+      if (stop > pages) {
+        newMaxPages = newMaxPages + 1 // add 1 back for missing '...' button
+        if (currPage.value === pages)
+          newMaxPages = newMaxPages + 1 // add another 1 back because 'next' button is hidden
+
+        stop = pages
+        start = Math.max(1, stop - newMaxPages)
+      }
+    }
+
+    // if we're on first page
+    if (currPage.value === 1) {
+      // add 1 more button to the end because 'prev' button is hidden, unless thay would exceed total pages
+      stop = Math.min(stop + 2, pages)
+    }
+
+    leftPages.value = []
+    for (let i = start; i <= stop; i++)
+      leftPages.value.push(i)
+  }
+}
+function setPaginationMaxPages(width: number) {
+  // fail gracefully with 10 as a the default
+  if (!initialCurrentPage || !pages)
+    return 10
+
+  // get width of buttons
+  const button = document.getElementsByClassName('pButton')[0]
+  const buttonWidth = Math.ceil(button.getBoundingClientRect().width)
+  const buttonMargin = getComputedStyle(button).marginRight
+  const itemWidth = Math.ceil(buttonWidth + (Number.parseInt(buttonMargin) * 2) + 1) // we add 1 to give us a little leeway
+  const prevButtonWidth = Math.ceil(document.getElementsByClassName('previous')[0].getBoundingClientRect().width + 10)
```

Comment:
use refs on previous and next buttons

const prevButtonWidth = prevButton.value ? Math.ceil(prevButton.value.getBoundingClientRect().width + 10) : 0
  const nextButtonWidth = nextButton.value ? Math.ceil(nextButton.value.getBoundingClientRect().width + 10) : 0
  
  create refs
  const prevButton = ref<HTMLElement | null>(null)
const nextButton = ref<HTMLElement | null>(null)

This approach keeps the component’s functionality encapsulated within Vue’s reactivity model, avoiding reliance on document, making the code cleaner and more maintainable.